jalib/jfilesystem.cpp

```diff
@@ -176,9 +176,11 @@ jalib::Filesystem::mkdir_r(const jalib::string &dir, mode_t mode)
 jalib::string
 jalib::Filesystem::GetProgramDir()
 {
-  static jalib::string value = DirName(GetProgramPath());
-
-  return value;
+  static jalib::string *value = NULL;
+  if (value == NULL) {
```

Comment:
Suggest adding comment:
// Technically, this is a memory leak, but `value` is static and so it happens only once.